src/microbe_stage/BackgroundPlane.cs

```diff
@@ -0,0 +1,263 @@
+
+        OnDisplayBackgroundParticlesChanged(Settings.Instance.DisplayBackgroundParticles);
+    }
+
+    public void SetWorldPosition(Vector2 position)
+    {
+        currentBackgroundMaterial.SetShaderParameter(worldPositionParameter, position);
+    }
+
+    public void SetVisibility(bool visible)
+    {
+        Visible = visible;
+
+        if (backgroundParticles != null)
+            OnDisplayBackgroundParticlesChanged(Settings.Instance.DisplayBackgroundParticles);
+    }
+
+    public void UpdateLightLevel(float lightLevel)
+    {
+        currentBackgroundMaterial.SetShaderParameter(lightLevelParameter, lightLevel);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            textureAlbedoParameter.Dispose();
+            blurAmountParameter.Dispose();
+            lightLevelParameter.Dispose();
+            distortionStrengthParameter.Dispose();
+            worldPositionParameter.Dispose();
+
+            if (blurPlanePath != null)
+            {
+                blurPlanePath.Dispose();
+                blurColorRectPath.Dispose();
+                backgroundPlanePath.Dispose();
+            }
+        }
+
+        base.Dispose(disposing);
+    }
+
+    private void OnDisplayBackgroundParticlesChanged(bool displayed)
+    {
+        if (backgroundParticles == null)
+        {
+            GD.PrintErr("MicrobeCamera didn't find background particles on settings change");
+            return;
+        }
+
+        backgroundParticles.Emitting = displayed;
+
+        if (displayed)
+        {
+            backgroundParticles.Show();
+        }
+        else
+        {
+            backgroundParticles.Hide();
+        }
+    }
+
+    private void OnBackgroundDistortionChanged(float value)
+    {
+        ApplyDistortionEffect();
+    }
+
+    private void ApplyDistortionEffect()
+    {
+        currentBackgroundMaterial.SetShaderParameter(distortionStrengthParameter,
+            Settings.Instance.MicrobeDistortionStrength.Value);
+    }
+
+    private void OnBackgroundBlurStrengthChanged(float value)
+    {
+        ApplyBlurEffect();
+    }
+
+    private void OnBackgroundBlurToggleChanged(bool value)
+    {
+        ApplyBlurEffect();
+    }
+
+    private void ApplyBlurEffect()
+    {
+        float blurStrength = Settings.Instance.MicrobeBackgroundBlurStrength;
+        bool enabled = blurStrength > 0 && Settings.Instance.MicrobeBackgroundBlurEnabled.Value;
+
+        if (enabled)
+        {
+            blurPlane.Visible = true;
+
+            canvasBlurMaterial.SetShaderParameter(blurAmountParameter, blurStrength);
+            spatialBlurMaterial.SetShaderParameter(blurAmountParameter, blurStrength);
+
+            if (backgroundPlane.GetParent() == this)
+            {
+                RemoveChild(backgroundPlane);
+                subViewport1.AddChild(backgroundPlane);
+            }
+
+            subViewport1.RenderTargetUpdateMode = SubViewport.UpdateMode.WhenVisible;
```

Comment:
I think disabling the update entirely and force enabling it when wanted would be better than this as that would logically avoid a check the engine needs to do to see if the viewport is actually visible in the scene tree each frame.